syntax = "proto3";

package gossip;

// Protobuf message for gossip communication
message GossipMessage {
  string message = 1;
  string sender_id = 2;
  int64 timestamp = 3;
  double latency_ms = 4;
  int32 round_count = 5;
}

message Acknowledgment {
  string details = 1;
}

// --- Added for One-Step Topology Update ---
message Neighbor {
  string pod_ip = 1;
  double weight = 2;
}

message NeighborList {
  repeated Neighbor neighbors = 1;
}

service GossipService {
  rpc SendMessage (GossipMessage) returns (Acknowledgment);
  // Accepts the list of neighbors directly instead of google.protobuf.Empty
  rpc UpdateNeighbors (NeighborList) returns (Acknowledgment);
}